DB/src/main/java/io/deephaven/db/v2/StreamTableToAppendOnlyTable.java

```diff
@@ -0,0 +1,82 @@
+package io.deephaven.db.v2;
+
+import io.deephaven.db.tables.Table;
+import io.deephaven.db.tables.utils.QueryPerformanceRecorder;
+import io.deephaven.db.v2.sources.ArrayBackedColumnSource;
+import io.deephaven.db.v2.sources.ColumnSource;
+import io.deephaven.db.v2.utils.ChunkUtils;
+import io.deephaven.db.v2.utils.Index;
+import io.deephaven.db.v2.utils.IndexShiftData;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Converts a Stream Table to an in-memory append only table.
+ *
+ * Note, this table will grow without bound as new stream values are encountered.
+ */
+public class StreamTableToAppendOnlyTable {
+    /**
+     * Convert a Stream Table to an in-memory append only table.
+     *
+     * @param streamTable the input stream table
+     * @return an append-only in-memory table representing all data encountered in the stream
+     */
+    public static Table streamToAppendOnlyTable(Table streamTable) {
+        return QueryPerformanceRecorder.withNugget("streamToAppendOnlyTable", () -> {
+            Object streamAttribute = streamTable.getAttribute(Table.STREAM_TABLE_ATTRIBUTE);
+            if (!(streamAttribute instanceof Boolean) || !(Boolean) streamAttribute) {
+                throw new IllegalArgumentException("Input is not a stream table!");
+            }
+
+            final Map<String, ArrayBackedColumnSource> columns = new LinkedHashMap<>();
+            final int columnCount = streamTable.getColumnSourceMap().size();
+            final ColumnSource [] sourceColumns = new ColumnSource[columnCount];
+            final ArrayBackedColumnSource [] destColumns = new ArrayBackedColumnSource[columnCount];
+            int colIdx = 0;
+            for (Map.Entry<String, ? extends ColumnSource> nameColumnSourceEntry : streamTable.getColumnSourceMap().entrySet()) {
+                final ColumnSource<?> existingColumn = nameColumnSourceEntry.getValue();
+                final ArrayBackedColumnSource<?> newColumn = ArrayBackedColumnSource.getMemoryColumnSource(0, existingColumn.getType(), existingColumn.getComponentType());
+                columns.put(nameColumnSourceEntry.getKey(), newColumn);
+                sourceColumns[colIdx] = existingColumn;
+                destColumns[colIdx++] = newColumn;
+            }
+
+            Index index = Index.FACTORY.getEmptyIndex();
```

Comment:
I was surprised that there was no initial copy here (and associated call to `ConstructSnapshot` or `checkInitiateTableOperations()`). It feels like it would be marginally more correct to capture that state, although irrelevant in the likely-common case of making one of these under the LTM lock soon after constructing a `StreamToTableAdapter`.